### ISSUE TITLE
Bug: `sort-enums`: Template literal expressions are not detected as dependencies

### ISSUE THREAD

**Issue by hugop95** [azat-io/eslint-plugin-perfectionist#264]
### Describe the bug

See code below

### Code example

```ts
enum Enum {
  B = 'B',
  A = `${B}`,
}
```
This code is considered invalid and requires `A` to be before `B` when it should not.

### ESLint version

8.57.0

### ESLint Plugin Perfectionist version

v3.5.0

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.
